rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -422,6 +425,38 @@ public void registerBtcTransaction(
         }
     }
 
+    private void updateSvpFundTransactionValuesIfPossible(BtcTransaction transaction) {
+        Optional<Sha256Hash> svpFundTransactionHashUnsigned = provider.getSvpFundTxHashUnsigned();
+        if (!svpFundTransactionHashUnsigned.isPresent()) {
+            return;
+        }
+
+        if (!isTheSvpFundTransaction(svpFundTransactionHashUnsigned.get(), transaction)) {
+            return;
+        }
+
+        updateSvpFundTransactionValues(transaction);
+    }
+
+    private boolean isTheSvpFundTransaction(Sha256Hash svpFundTransactionHashUnsigned, BtcTransaction transaction) {
+        if (!transaction.hasWitness()) {
+            BtcTransaction transactionCopy = new BtcTransaction(networkParameters, transaction.bitcoinSerialize()); // this is needed to not remove signatures from the actual tx
+            BitcoinUtils.removeSignaturesFromTransactionWithP2shMultiSigInputs(transactionCopy);
+            return transactionCopy.getHash().equals(svpFundTransactionHashUnsigned);
+        }
+
+        return transaction.getHash().equals(svpFundTransactionHashUnsigned);
+    }
```

Comment:
```suggestion
    private void updateSvpFundTransactionValuesIfPossible(BtcTransaction transaction) {
        if (!isTheSvpFundTransaction(transaction)) {
            return;
        }

        updateSvpFundTransactionValues(transaction);
    }

    private boolean isTheSvpFundTransaction(BtcTransaction transaction) {
        Optional<Sha256Hash> svpFundTransactionHashUnsigned = provider.getSvpFundTxHashUnsigned();
        if (!svpFundTransactionHashUnsigned.isPresent()) {
            return false;
        }
        
        Sha256Hash txHash = transaction.getHash();
        if (!transaction.hasWitness()) {
            BtcTransaction transactionWithoutSignatures = new BtcTransaction(networkParameters, transaction.bitcoinSerialize()); // this is needed to not remove signatures from the actual tx
            BitcoinUtils.removeSignaturesFromTransactionWithP2shMultiSigInputs(transactionCopy);
            txHash = transactionWithoutSignatures.getHash();
        }

        return txHash.equals(svpFundTransactionHashUnsigned);
    }
```

I think that simplifies things a bit. Only get the fund tx from storage where needed. And only do the hash comparisson once.

Reply:
i don't super like the `isTheSvpFundTransaction` method also doing the `svpFundTxHashUnsigned` check.
And overall remember it would be removed soon and added to method being called from `updateCollections`. wdyt?

Reply:
btw, the other suggestion about the txHash was done :) Thanks